docs/docs/api-files-gatsby-browser.md

```diff
@@ -0,0 +1,25 @@
+---
+title: The gatsby-browser.js API file
+---
+
+The file `gatsby-browser.js` lets you respond to actions within the browser, and wrap your site in additional components. The [Gatsby Browser API](/docs/browser-apis) gives you many options for interacting with the client side of Gatsby.
+
+The APIs `wrapPageElement` and `wrapRootElement` exist in both the browser and [SSR APIs](/docs/ssr-apis). If you use one of them, consider if you should implement it in both `gatsby-ssr.js` and `gatsby-browser.js`.
```

Comment:
Maybe explain why I'd want it in both, or not want it in both.

Reply:
Totes! Added in https://github.com/gatsbyjs/gatsby/pull/17745/commits/5262dda317c8db4f1879c686d4dfc7c290da845b

Reply:
The new wording is much more confusing :(

Reply:
I would stick with the old wording but explain:
To be equal with the ssr generated pages after hydration on the browser.

Reply:
Noted, I think my biggest challenges with this was that I needed to make sure Provider components were wrapped in both Browser and SSR so that things like page components rendered correctly... is there an existing doc page we could just link to about this?

How about:

The APIs `wrapPageElement` and `wrapRootElement` exist in both the browser and [SSR APIs](/docs/ssr-apis). If you use one of them, consider if you should implement it in both `gatsby-ssr.js` and `gatsby-browser.js` so that SSR generated pages are the same after hydrated in the browser.

Reply:
looks better this way you just mentioned.

Reply:
Rewrote in https://github.com/gatsbyjs/gatsby/pull/17745/commits/15296e37e8bfc664a742666ec3f8ea1a73254077

Reply:
But with youre wording I understand that only with the `wrapPageElemenr` and the other explicit named I need to put it in both. 

But I have to put all into both. 

Also as first time user I don't know.what are the provider/wrap/others - and if I not use this special names I have to put only in one.

I would word like: you have to check always to put it in both 
and then something like:  also don't forget it for the provider/wrap/others

Reply:
I call for review from @marcysutton - she is always very good in wording complicated things in good words :)

Reply:
Agreed - I think some APIs have more "weight" than others in terms of impact across SSR/browser.

Reply:
We could also just remove this language in these API file docs. If a user is actually going to be looking into implementing the wrapper APIs, the docs for both clearly state that they should probably implement them in Browser and SSR.